basic-application/basic-application-webapp/src/main/java/fr/openwide/core/basicapp/web/application/BasicApplicationApplication.java

```diff
@@ -105,7 +110,9 @@ protected IConverterLocator newConverterLocator() {
 		converterLocator.set(TechnicalUser.class, UserRenderer.get());
 		converterLocator.set(BasicUser.class, UserRenderer.get());
 		converterLocator.set(UserGroup.class, UserGroupRenderer.get());
+		converterLocator.set(GenericLocalizedGenericListItem.class, LocalizedGenericListItemRenderer.get());
```

Comment:
We can, but unfortunately I doubt it will result in anything.

## Why it's not easy

There are several problems with this, the main (and blocking) one being that `IConverter<T>.convertToObject(String)` is covariant in `T` :

 * In our case, it does make sense to support inheritance because we only perform a one-way conversion with `convertToString` (`T2 extends T1` => `String`), so the concrete subclass `T2` of `T1` doesn't matter : the converter only needs to support `T1`.
 * In the case of `convertToObject` (`String` => `T2 extends T1`) conversions, though, the converter should return the exact type `T2`, so it **must** explicitly support `T2`.

To sum up, if converter registration was to support inheritance, then each converter of `T` to/from `String` should support `T`, but also support explicitly each and every subclass of `T`.

## Why I think it's no Wicket business

Now, the whole thing wouldn't be an issue if either:

 1. there was some way to declare which subtype of `T` the converters support
 2. or the `IConverterLocator` could provide the required type to the converter

The first option is sort of already supported by wicket's implementation of `IConverterLocator` (`org.apache.wicket.ConverterLocator`): it's sort of what we do when we explicitly declare a single converter for each subtype of `T`.

The second option would, in fact, make the whole conversion system quite complex, but very close to what Spring provides. So I think that wiring our own implementation of `IConverter` to an instance of Spring's `ConversionService` would do the trick, provided we use Spring's way of dealing with locale (the `LocaleContextHolder`). We could even support registration of Wicket's converters. I'd be surprised if there wasn't already such an implementation out in the wild (in wicketstuff, maybe?).

To sum up, I think the answer we'll get will be: "This would require either a deep change in the converter API (to add the required type as an argument) or adding converter interfaces and implementing a complex mechanism that's already available elsewhere (e.g. in Spring). In either case, it's not gonna happen."

## Some more constructive advice

As I said above, we could just wrap some instance of Spring `ConversionService` so as to delegate the whole converter registration/retrieval process. Some (minor) work would need to be done regarding the locale stuff (with `LocaleContextHolder`), and regarding the registration of non-spring interfaces (IConverter). But overall I think it'd be quite a minor change compared to what we (or Wicket's staff) would need to do to support inheritance from scratch.